### ISSUE TITLE
Failed to set thread priority - Ubuntu 20.04

### ISSUE THREAD

**Issue by Acetsetebos** [ValveSoftware/steam-for-linux#7193]
#### Your system information

* Steam client version (build number or date): updater built Jun  4 2020 05:50:42
* Distribution (e.g. Ubuntu): Xubuntu 20.04
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
I installed steam through my distro's package manager and it was working just fine for awhile, but now when I try to open it I get this error:

Running Steam on ubuntu 20.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Steam client's requirements are satisfied
/home/caleb/.steam/debian-installation/ubuntu12_32/steam -nominidumps -nobreakpad
[2020-06-13 11:21:19] Startup - updater built Jun  4 2020 05:50:42
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2020-06-13 11:21:21] Checking for update on startup
[2020-06-13 11:21:21] Checking for available updates...
[2020-06-13 11:21:21] Downloading manifest: client-download.steampowered.com/client/steam_client_ubuntu12
[2020-06-13 11:21:21] Download skipped: /client/steam_client_ubuntu12 version 1591251555, installed version 1591251555
[2020-06-13 11:21:21] Nothing to do
[2020-06-13 11:21:21] Verifying installation...
[2020-06-13 11:21:21] Performing checksum verification of executable files
[2020-06-13 11:21:26] Verification complete
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
Warning: failed to set thread priority: set failed for 4: -1: setpriority() failed
Warning: failed to set thread priority: set failed for priority 4
Warning: support for elevated priorities is most likely unavailable, suppressing future warnings

#### Steps for reproducing this issue:

1. `sudo apt install steam-installer`
2. run steam
3. `¯\_(ツ)_/¯ `

**Comment by kisak-valve** [comment]
Hello @Acetsetebos, failing to set thread priority is a red herring. What video card / driver version are you using?

**Comment by Acetsetebos** [comment]
@kisak-valve 
VGA compatible controller: Intel Corporation UHD Graphics 620 (rev 07)
driver = i915

**Comment by kisak-valve** [comment]
Do you have a bunch of games installed? (hundreds)

If Steam is sitting there stalled without a fatal error, you may be encountering #7117.

**Comment by Acetsetebos** [comment]
I had less than ten games installed. Also steam doesn't stall when I launch it, it crashes immediately after the connecting to account window closes.

**Comment by kisak-valve** [comment]
There's no fatal error in the terminal spew in the opening post, implying that Steam either stalled or is running in the background.

Can you check if it is running in the background? Worse case scenario you can kill it with `pkill -9 steam`. After that, if you run steam from a terminal, does it give the same output and come back to a command prompt or just sit there?

**Comment by Acetsetebos** [comment]
I checked my gui task manager and didn't see any steam related processes running.
I ran `pkill -9 steam` just in case then launched steam in terminal, and I got the same crash and error message, after the crash the terminal's command prompt pops back up.

**Comment by kisak-valve** [comment]
Can you check if the distro-provided package is redirecting any useful hints to `~/.steam/error.log`?

**Comment by Acetsetebos** [comment]
Here's the contents of the error.log file https://gist.github.com/Acetsetebos/2d7beb8113136bdc5e12258d1434b437
The line at the very bottom reads: `/home/caleb/.steam/debian-installation/steam.sh: line 750:  2680 Segmentation fault      (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"` and I assume that has something to do with the crash.

**Comment by TTimo** [comment]
I see `steam -nominidumps -nobreakpad` - looks like distro modified package. Consider replacing the distro version of steam with the official package - see http://repo.steampowered.com/steam/ for details.

**Comment by Acetsetebos** [comment]
Solved!
After installing the .deb package from the page @TTimo linked I was still getting the same error as before, but the error now contained the line `vulkan: No DRI3 support`
So I added this line to `/etc/X11/xorg.conf.d/20-intel.conf`:
```
Section "Device"
   Identifier  "Intel Graphics"
   Driver      "intel"
   Option      "DRI"    "3"
EndSection
```
And after a restart, steam works just fine.

**Comment by KaranX98** [comment]
kxrxn98@pop-os:~$ steam
Running Steam on pop 20.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Steam client's requirements are satisfied
/home/kxrxn98/.steam/debian-installation/ubuntu12_32/steam -nominidumps -nobreakpad
[2020-08-31 16:01:35] Startup - updater built Aug 30 2020 20:57:23
[2020-08-31 16:01:36] Verifying installation...
[2020-08-31 16:01:36] Verification complete
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
Warning: failed to set thread priority: set failed for 4: -1: setpriority() failed
Warning: failed to set thread priority: set failed for priority 4
Warning: support for elevated priorities is most likely unavailable, suppressing future warnings
Warning: failed to set thread priority: set failed for 4: -1: setpriority() failed
Warning: failed to set thread priority: set failed for priority 4
Warning: support for elevated priorities is most likely unavailable, suppressing future warnings
[2020-08-31 16:02:20] Shutdown

Any help would be appriciated

**Comment by TTimo** [comment]
Please see https://github.com/ValveSoftware/steam-for-linux/issues/7193#issuecomment-645666757 - use the official Steam packaging that does not disable crash reporting.

**Comment by KaranX98** [comment]
> Please see [#7193 (comment)](https://github.com/ValveSoftware/steam-for-linux/issues/7193#issuecomment-645666757) - use the official Steam packaging that does not disable crash reporting.

Yup, that doesn't work for me.

**Comment by TTimo** [comment]
Right, but it should offer more information about what is failing rather than silently shutdown. Do you get a crash dump and a crash upload ID?

**Comment by KaranX98** [comment]
I installed linuxmint because it showed a few extra errors about gtk 2.0. While it removed the gtk errors the setpriority() problem still persisted so I used the method from #7193 (https://github.com/ValveSoftware/steam-for-linux/issues/7193#issuecomment-645666757) but now the gtk errors came back, so I reverted back to the distro version for steam. The main issue is setpriority() error, while steam launches correctly but I cannot run any games with proton.

**Comment by KaranX98** [comment]
> Right, but it should offer more information about what is failing rather than silently shutdown. Do you get a crash dump and a crash upload ID?

I installed linuxmint because it showed a few extra errors about gtk 2.0. While it removed the gtk errors the setpriority() problem still persisted so I used the method from #7193 (#7193 (comment)) but now the gtk errors came back, so I reverted back to the distro version for steam. The main issue is setpriority() error, while steam launches correctly but I cannot run any games with proton.
